test.py

```diff
@@ -28,6 +28,34 @@ def test_deparam(self):
             good = base + good
             self.assertEqual(url.parse(bad).deparam(['c']).utf8(), good)
 
+    def test_deparam_case_insensitivity(self):
+        '''Parameters to remove can be supplied in any case.'''
+        examples = [
+            ('?hELLo=2', ''),
+            ('?HELLo=2', '')
+        ]
+        base = 'http://testing.com/page'
+        for bad, good in examples:
+            bad = base + bad
+            good = base + good
+            self.assertEqual(url.parse(bad).deparam(['HeLlO']).utf8(), good)
+
+    def test_filter_params(self):
+        '''We can filter parameters.'''
+        def function(name, value):
```

Comment:
The only change I might make is to supply more context to the passed predicate function.  Right now it only sees name-value pairs in isolation.  This makes it hard to enforce constraints depending on the parameters as a set, such as "only keep up to the first three parameters."  By giving more context to the function, it can gain the ability to decide these constraints.

The extra context might be just the index (position) of the name-value pair as it appears in the url, or perhaps a `dict` (or other container) describing all pairs in the url.  Or both.